### PR TITLE
Migrate to npm scripts

### PR DESCRIPTION
**Pull request by ryanggrey**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "presets": [
+        "es2015",
```
Comment (oriondean) [inline .babelrc]: How about using the [env plugin](https://babeljs.io/docs/plugins/preset-env/)?

```
"presets": [
    ["env", {
      "targets": {
        "chrome": "last 2 versions",
        "ie": 10
      }
    }]
  ]
```

File: package.json [hunk 2]
```diff
@@ -16,28 +16,28 @@
...
+    "uglify-js": "^3.0.24",
     "jest": "^20.0.4",
     "selenium-webdriver": "^3.4.0",
     "webdriver-manager": "^12.0.6"
   },
   "scripts": {
+    "clean": "rimraf ./dist && mkdirp ./dist/",
+    "compile": "browserify -d ./src/finput.js -s finput -o ./dist/finput.js -t babelify && uglifyjs ./dist/finput.js -o ./dist/finput.min.js",
+    "build": "npm run clean && npm run compile",
+    "start": "budo -l -o -p 3000 src/finput.js:dist/finput.js -- -s finput -t babelify",
```
Comment (oriondean) [inline package.json]: Could we add ` "postinstall": "webdriver-manager update" ` too

File: package.json [hunk 3]
```diff
@@ -16,28 +16,28 @@
...
+    "jasmine": "^2.4.1",
+    "mkdirp": "^0.5.1",
+    "rimraf": "^2.6.1",
+    "uglify-js": "^3.0.24",
     "jest": "^20.0.4",
     "selenium-webdriver": "^3.4.0",
     "webdriver-manager": "^12.0.6"
   },
   "scripts": {
+    "clean": "rimraf ./dist && mkdirp ./dist/",
```
Comment (oriondean) [inline package.json]: You should be able to shorten this to just `rimraf ./dist/*`

Comment (ryanggrey) [inline package.json]: I think I might need both. `browserify` doesn't create paths that don't exist (which surprised me). So the case where `npm run build` is running for the very first time requires that something creates the path.

Comment (oriondean) [inline package.json]: rimraf ./dist/* won't delete the dist directory so you shouldn't need to create any paths afterwards

File: package.json (hunk 2 above)
Comment (ryanggrey) [inline package.json]: Yeah, I'd been considering something like that so will take a look at adding. I think I can add the start and stop of the webdriver server to our `test` script too to take away the need to manually do anything with `webdriver-manager`.

File: .babelrc (hunk 1 above)
Comment (ryanggrey) [inline .babelrc]: There's a couple of things that make me unsure about making this change:

**Version Ranges**
We will no longer explicitly control the version of JS that we target. New releases of browsers could change our outputted bundle between our releases without us realising. This is possibly ok if we're informing users of the version range in our docs, but see point below. However, it's true that the environment plugin allows us to specify specific versions, so perhaps this isn't a problem.

**Browser Support: Versions vs Features**
We recently discussed that informing users of the js/browser features that we require was preferred over stating browser version numbers (see [Required Browser Features)](https://github.com/ScottLogic/finput#required-browser-features). With that in mind I prefer seeing exactly which version of JS we're targeting rather than which browser versions.

File: package.json (hunk 3 above)
Comment (ryanggrey) [inline package.json]: `browserify` doesn't create paths that don't exist. So something will need to create the `./dist/` directory on first run.

File: package.json (hunk 2 above)
Comment (ryanggrey) [inline package.json]: Turns out that starting webdriver in the background as part of the `test` script is going to be difficult due to - https://github.com/angular/webdriver-manager/issues/212. So I'll just pushup the `postinstall` change for now.
